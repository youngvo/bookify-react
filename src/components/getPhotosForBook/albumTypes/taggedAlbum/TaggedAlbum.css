.tagged-album {
    height: 70px;
    width: 400px;
    background-image: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
    border-radius: 8px;
    display: inline-flex;
    position: absolute;
    box-shadow: 2px 2px 2px #888;
    cursor: pointer;
}

.inner-shadow-tagged-album {
    box-shadow: inset 2px 2px 2px rgba(150, 150, 150, 0.75);
    height: 67px;
    width: 397px;
    background-image: transparent;
    border-radius: 5px;
}

.outer-shadow-tagged-album {
    border: 3px solid #EC6E1A;
    box-shadow: 2px 2px 1px rgba(150, 150, 150, 0.75);
    height: 67px;
    width: 397px;
    background-image: transparent;
    border-radius: 8px;
}

.tagged-album:hover {
    background:#a9e1fa;
}

.title-tagged-album {
    width: 40%;
    height: 100%;
    float:left;
}

.title-tagged-album div {
    position: relative;
    height: 25px;
    font-size: 14px;
    top: 55%;
    transform: translate(0, -50%);
    color:#555;
    left:7px;
}

.view-info-album-zone * {
    padding: 0;
    margin: 2px 0 0 8px;
    color: dimgray;
}

.name-album {
    font-size: 15px;
}

.amount-of-photo {
    font-size: 12px;
}

.tagged-album .preview-album-zone {
    height: 100% !important;
    width: 60%;
    margin: 0 !important;
    padding: 0 !important;
    overflow: hidden;
    float:left;
}

.tagged-album .image-preview-tagged-album {
    height: 50px;
    width: 50px;
    background-color: burlywood;
    position: relative;
    margin-left: 5px;
    top: 50%;
    transform: translate(0, -50%);
    overflow: hidden;
    float:left;
}

.tagged-album .image-preview-tagged-album img{
    max-height: 100px;
    min-width: 50px;
    position: absolute;

}

.content-left-list-album{
    position: relative;
    width:100%;
    height:74%;
}