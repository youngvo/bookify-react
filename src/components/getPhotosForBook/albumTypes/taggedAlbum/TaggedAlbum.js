import React , {Component} from 'react';
import LocaleUtils from './../../../../utils/LocaleUtils';
import './TaggedAlbum.css';

class TaggedAlbum extends Component{
    renderPictures (picUrls) {
        let pics = picUrls.map((picUrl, index) => (
            <div className="image-preview-tagged-album">
                <img className="" key={index} src={picUrl} alt="Tagged Album" />
            </div>
        ));

        return (<div className="preview-album-zone" >{pics}</div>);
    }

    renderCountPhotosSelected = (photoSelectedCount) => {
        if (photoSelectedCount && photoSelectedCount > 0) {
            // return (<div>{photoSelectedCount}</div>);
            return (
                <div className="badge-notification">{photoSelectedCount}</div>
            );
        }
    }

    onTaggedAlbumClick(album, onClick) {
        if (onClick) onClick(album);
    }

    render() {
        const { album, photoSelectedCount, onClick } = this.props;

        return (
            <div className="tagged-album" onClick={this.onTaggedAlbumClick.bind(this, album, onClick)}>
                <div className={ photoSelectedCount > 0 ? "outer-shadow-tagged-album" : "tagged-album"}>
                    <div className={ photoSelectedCount > 0 ? "inner-shadow-tagged-album" : "tagged-album"}>
                        {this.renderCountPhotosSelected(photoSelectedCount)}
                        <div className="title-tagged-album">
                        <div>{LocaleUtils.instance.translate('import.facebook.album.tagged')}</div>
                        </div>
                        {this.renderPictures(album.picUrls)}
                    </div>
                </div>
            </div>
        );
    }
}

export default TaggedAlbum;
