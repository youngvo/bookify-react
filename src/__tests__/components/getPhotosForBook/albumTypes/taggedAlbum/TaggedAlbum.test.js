import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TaggedAlbum from './../../../../../components/getPhotosForBook/albumTypes/taggedAlbum/TaggedAlbum';

describe('Test TaggedAlbum Component', () => {
    let album = {
        name: 'album',
        photoIds: [],
        picUrls: ["http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html","http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html"],
    };
    let photoSelectedCount = 0; 
    let onClick;
    let taggedAlbum;
    beforeEach(() => {
        onClick = jest.fn();
        taggedAlbum = shallow(<TaggedAlbum album={album} photoSelectedCount={photoSelectedCount} onClick={onClick} />);
    })

    describe('TaggedAlbum Snapshot', () => {
        it('TaggedAlbum does not change UI ', () => {
            const component = renderer.create(<TaggedAlbum album={album} photoSelectedCount={photoSelectedCount} onClick={onClick} />);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('TaggedAlbum test DOM', () => {
        it('contains div with className=\"tagged-album\" ', () => {
            expect(taggedAlbum.find('div.tagged-album').length).toEqual(1);
            expect(taggedAlbum.find('div.tagged-album').text()).toEqual("Photos you are tagged ");
        })

        describe('test renderCountPhotosSelected() return', () => {
            it('return div element', () => {
                taggedAlbum.setProps({photoSelectedCount: 10});
                expect(taggedAlbum.find('div').first().text()).toEqual("10Photos you are tagged ");
            });
        });

        describe('test renderPictures() return', () => {
            it('return div element with className="preview-album-zone" ', () => {
                expect(taggedAlbum.find('.preview-album-zone').length).toEqual(1);
            });

            it('return div contains <img/> tag = picUrl.lenght param ', () => {
                expect(taggedAlbum.find('img.image-preview-tagged-album').length).toEqual(album.picUrls.length);
            });
        })
    });

    describe('TaggedAlbum test function', () => {
        it('TaggedAlbum dispatch onClick event when clicked', () => {
            taggedAlbum.simulate('click');
            expect(onClick.mock.calls.length).toEqual(1);
            taggedAlbum.simulate('click');
            expect(onClick.mock.calls.length).toEqual(2);
            taggedAlbum.simulate('click');
            expect(onClick.mock.calls.length).not.toEqual(4);
        })
    });



});