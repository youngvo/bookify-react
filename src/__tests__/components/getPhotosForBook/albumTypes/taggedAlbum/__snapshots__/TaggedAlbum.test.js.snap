exports[`Test TaggedAlbum Component TaggedAlbum Snapshot TaggedAlbum does not change UI  1`] = `
<div
  className="tagged-album"
  onClick={[Function]}>
  <div
    className="title-tagged-album">
    <div>
      Photos you are tagged
    </div>
  </div>
  <div
    className="preview-album-zone">
     
    <img
      alt="Tagged Album"
      className="image-preview-tagged-album"
      src="http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html" />
    <img
      alt="Tagged Album"
      className="image-preview-tagged-album"
      src="http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html" />
     
  </div>
</div>
`;
